Change aht-dummy rpm test

aaradhak opened the pull request.

dustymabe commented on mantle/kola/tests/ostree/unlock.go:
```diff
@@ -44,8 +44,8 @@ func init() {
-	rpmUrl  string = "https://raw.githubusercontent.com/projectatomic/atomic-host-tests/master/rpm/aht-dummy-1.0-1.noarch.rpm"
-	rpmName string = "aht-dummy"
+	rpmUrl  string = "https://dl.fedoraproject.org/pub/epel/"
+	rpmName string = " epel-release-latest-9"
```
The name of the rpm is `epel-release`:

```
$ rpm -qip epel-release-latest-9.noarch.rpm 
warning: epel-release-latest-9.noarch.rpm: Header V4 RSA/SHA256 Signature, key ID 3228467c: NOKEY
Name        : epel-release
Version     : 9
Release     : 10.el9
Architecture: noarch
Install Date: (not installed)
Group       : Unspecified
Size        : 26854
License     : GPLv2
Signature   : RSA/SHA256, Tue 22 Apr 2025 02:57:52 PM EDT, Key ID 8a3872bf3228467c
Source RPM  : epel-release-9-10.el9.src.rpm
Build Date  : Tue 22 Apr 2025 01:19:57 PM EDT
Build Host  : buildvm-a64-14.iad2.fedoraproject.org
Packager    : Fedora Project
Vendor      : Fedora Project
URL         : http://download.fedoraproject.org/pub/epel
Bug URL     : https://bugz.fedoraproject.org/epel-release
Summary     : Extra Packages for Enterprise Linux repository configuration
Description :
This package contains the Extra Packages for Enterprise Linux (EPEL) repository
GPG key as well as configuration for yum.
```

aaradhak replied:
I have updated it

dustymabe commented on mantle/kola/tests/rpmostree/deployments.go:
```diff
@@ -49,14 +49,14 @@ func init() {
+					"source": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm",
 					"verification": {
-					  "hash": "sha512-da29ae637b30647cab2386a2ce6b4223c3ad7120ae8dd32d9ce275f26a11946400bba0b86f6feabb9fb83622856ef39f8cecf14b4975638c4d8c0cf33b0f7b26"
+					  "hash": "sha512-265ab9d1e4a5dd1b0277b9c014fa72ba4cced7bad2796d89a89b303bb233545dc9fe41917da6339c60776b1abb0281317ebea760c0f8e70916f73830d0eca10f"
```
we might have to update this periodically. Maybe it would be better to not include it?

@jlebon WDYT?

jlebon replied:
I don't think that aspect is relevant to the test, so SGTM.

dustymabe replied:
@aaradhak - drop the `verification` section here since we don't want this to fail if the RPM gets updated.

jlebon commented on mantle/kola/tests/rpmostree/deployments.go:
```diff
@@ -49,14 +49,14 @@ func init() {
 			"storage": {
 			  "files": [
 				{
-				  "path": "/var/home/core/aht-dummy.rpm",
```
I think other things need updated in this test. Search for `aht-dummy.rpm`.

dustymabe commented on mantle/kola/tests/rpmostree/deployments.go:
```diff
@@ -190,9 +190,9 @@ func rpmOstreeUpgradeRollback(c cluster.TestCluster) {
-	var ahtRpmPath = "/var/home/core/aht-dummy.rpm"
-	var installPkgName = "aht-dummy-1.0-1.noarch"
-	var installBinName = "aht-dummy"
+	var epelRpmPath = "/var/home/core/epel-release-latest-9.noarch.rpm"
```
we need to update the `sudo rpm-ostree install` below to use this variable.

aaradhak replied:
I have updated it

dustymabe commented on mantle/kola/tests/ostree/unlock.go:
```diff
@@ -44,8 +44,8 @@ func init() {
-	rpmName string = "aht-dummy"
+	rpmUrl  string = "https://dl.fedoraproject.org/pub/epel/"
+	rpmName string = "epel-release"
 )
```
you'll probably need to remove the `command -v` test below because this RPM doesn't provide any binaries.

aaradhak replied:
removed the unnecessary binary related variables

dustymabe commented on mantle/kola/tests/rpmostree/deployments.go:
```diff
@@ -190,9 +190,9 @@ func rpmOstreeUpgradeRollback(c cluster.TestCluster) {
-	var installBinName = "aht-dummy"
+	var epelRpmPath = "/var/home/core/epel-release-latest-9.noarch.rpm"
+	var installPkgName = "epel-release-latest-9.noarch.rpm"
+	var installBinName = "epel-release"
```
This package doesn't provide any binaries so we can drop all the binary tests (i.e. `command -v`).

aaradhak replied:
I have dropped the binary related variables

dustymabe commented on mantle/kola/tests/rpmostree/deployments.go:
```diff
@@ -190,9 +190,9 @@ func rpmOstreeUpgradeRollback(c cluster.TestCluster) {
-	var installPkgName = "aht-dummy-1.0-1.noarch"
-	var installBinName = "aht-dummy"
+	var epelRpmPath = "/var/home/core/epel-release-latest-9.noarch.rpm"
+	var installPkgName = "epel-release-latest-9.noarch.rpm"
```
```suggestion
	var installPkgName = "epel-release"
```

The name of the RPM is `epel-release`

aaradhak commented:
CI error - 

```
[2025-05-14T21:04:54.608Z]     --- FAIL: ostree.unlock/install (0.20s)

[2025-05-14T21:04:54.608Z]             cluster.go:151: warning: /var/tmp/rpm-tmp.u8TtqM: Header V4 RSA/SHA256 Signature, key ID 3228467c: NOKEY

[2025-05-14T21:04:54.608Z]             cluster.go:151: error: Failed dependencies:

[2025-05-14T21:04:54.608Z]             cluster.go:151: 	redhat-release >= 9 is needed by epel-release-9-10.el9.noarch

[2025-05-14T21:04:54.608Z]             cluster.go:151: 	fedora-release conflicts with epel-release-9-10.el9.noarch

[2025-05-14T21:04:54.608Z]             unlock.go:147: Failed to install RPM: Process exited with status 1
```

aaradhak commented:
I dont think [epel-release-latest-8.noarch.rpm](https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm) makes a difference here. 


```
[2025-05-14T22:13:10.481Z]     --- FAIL: ostree.unlock/install (0.23s)

[2025-05-14T22:13:10.481Z]             cluster.go:151: warning: /var/tmp/rpm-tmp.qPkyTE: Header V4 RSA/SHA256 Signature, key ID 2f86d6a1: NOKEY

[2025-05-14T22:13:10.481Z]             cluster.go:151: error: Failed dependencies:

[2025-05-14T22:13:10.481Z]             cluster.go:151: 	redhat-release >= 8 is needed by epel-release-8-22.el8.noarch

[2025-05-14T22:13:10.481Z]             cluster.go:151: 	fedora-release conflicts with epel-release-8-22.el8.noarch

[2025-05-14T22:13:10.481Z]             unlock.go:147: Failed to install RPM: Process exited with status 1
```

dustymabe commented on mantle/kola/tests/ostree/unlock.go:
```diff
@@ -44,8 +44,8 @@ func init() {
 var (
-	rpmUrl  string = "https://raw.githubusercontent.com/projectatomic/atomic-host-tests/master/rpm/aht-dummy-1.0-1.noarch.rpm"
-	rpmName string = "aht-dummy"
+	rpmUrl  string = "https://mirrors.rpmfusion.org/free/fedora/rpmfusion-free-release-rawhide.noarch.rpm"
```
You can't just use the rawhide RPM on F42 (testing-devel) so you'd have to conditionalize everything. I'd say at this point let's just try to build the dummy rpm again in a new dedicated repo for that purpose.

aaradhak commented:
With the rpm-fusion-rawhide pkg , looks like it it is dependent on another pkg. The kola tests failed because it attempted to install rpm-fusion package that has a dependency `system-release(43)`

dustymabe commented:
> With the rpm-fusion-rawhide pkg , looks like it it is dependent on another pkg. The kola tests failed because it attempted to install rpm-fusion package that has a dependency `system-release(43)`

Yes. By design the current `rawhide` package will only work on Fedora 43 (rawhide is currently Fedora 43).

aaradhak commented:
> > With the rpm-fusion-rawhide pkg , looks like it it is dependent on another pkg. The kola tests failed because it attempted to install rpm-fusion package that has a dependency `system-release(43)`
> 
> Yes. By design the current `rawhide` package will only work on Fedora 43 (rawhide is currently Fedora 43).

Do we need to look for a pkg without any dependency?

dustymabe commented:
> Do we need to look for a pkg without any dependency?

Either that (i.e. find a package that works everywhere, which will probably only find if we build it) or dynamically update the test based on the target we are testing against.

At this point I'm voting we revive the dummy package.

aaradhak commented:
ok, I will look into adding the aht-dummy pkg by creating a new repo.

dustymabe commented:
> Created a new [aht-dummy-repo](https://github.com/coreos/aht-dummy-rpm) and updated the aht-dummy-rpm build

Sorry to bikeshed on the name.. Could we give it a more descriptive name now?

- `aht` stands for "Atomic Host Tests" and Atomic Host is dead, so the name doesn't make much sense anymore either.
- The word `repo` could be confusing too, because this doesn't actually host a yum repo. It's just the RPM file.

WDYT about renaming the git repo (which should be easy to do) to: `coreos-dummy-rpm`?

aaradhak commented:
> > Created a new [aht-dummy-repo](https://github.com/coreos/aht-dummy-rpm) and updated the aht-dummy-rpm build
> 
> Sorry to bikeshed on the name.. Could we give it a more descriptive name now?
> 
> * `aht` stands for "Atomic Host Tests" and Atomic Host is dead, so the name doesn't make much sense anymore either.
> * The word `repo` could be confusing too, because this doesn't actually host a yum repo. It's just the RPM file.
> 
> WDYT about renaming the git repo (which should be easy to do) to: `coreos-dummy-rpm`?

Sure, I can make that change in the name

aaradhak commented:
@dustymabe I have updated the repo name ro coreos-dummy-rpm https://github.com/coreos/coreos-dummy-rpm , I have created a release tag too for this - https://github.com/coreos/coreos-dummy-rpm/releases/tag/v1.0.0

dustymabe commented:
> @dustymabe I have updated the repo name ro coreos-dummy-rpm https://github.com/coreos/coreos-dummy-rpm , I have created a release tag too for this - https://github.com/coreos/coreos-dummy-rpm/releases/tag/v1.0.0

Nice. So here's what I would do next:

- rename the actual spec file (and also the name inside the spec file) to `coreos-dummy` instead of `aht-dummy`.

The following could be done, but it's optional as it provides only marginal benefit/gain:

- delete the actual *rpm file from git
- commit that and tag it with `v2.0.0`
- build the RPM and then attach the `.rpm` file to the `v2.0.0` release (i.e. edit the release and attach binaries to it).

aaradhak commented:
I have updated the spec file to carry the name coreos-dummy instead of aht-dummy. I have also built the rpm again and updated the release to v2.0.0 - https://github.com/coreos/coreos-dummy-rpm/releases/tag/V2.0.0

dustymabe reviewed (approved):
LGTM
